### DISCUSSION TITLE
工作流实例中重新运行某个shell任务，提示java.lang.ArrayIndexOutOfBoundsException: 1

### DISCUSSION THREAD

**Discussion by zxcseraph** [apache/dolphinscheduler#11792] [Q&A]
在一个工作流中有多个互相依赖的脚本，其中一个运行的有些问题，点击重新运行的时候，会提示失败，观察日志有如下内容。主要是提示java的索引问题，但是我用的是shell脚本，所以无从下手，谁碰见过类似问题受累给提示一下，谢谢。

[LOG-PATH]: /home/dolphinscheduler/worker-server/logs/20220906/6773442559590_13-341-617.log, [HOST]:  Host{address='172.17.0.1:1234', ip='172.17.0.1', port=1234}
[INFO] 2022-09-06 04:50:45.838 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[84] - shell task params {"localParams":[{"prop":"ds_ds_run","direct":"IN","type":"VARCHAR","value":"${system.datetime}"},{"prop":"mgr_out_report","direct":"OUT","type":"VARCHAR","value":""},{"prop":"continue_flag","direct":"OUT","type":"VARCHAR","value":""}],"rawScript":"set -x\r\nif [ X\"${start_flag}\" != Xstart ] \r\nthen\r\n    if [ X\"${continue_flag}\" != Xcontinue ]\r\n    then\r\n        global_report=\"${mgr_out_report}\"\r\n        echo '${setValue(mgr_out_report='${global_report}')}'\r\n        exit 0\r\n    else\r\n        global_report=\"\"\r\n    fi\r\nfi\r\nbasedir=${global_basedir}\r\nsource ./public.sh\r\nsource ./alarm/testInMGR.sh\r\nmv ${global_basedir}/eblogs/report.log  ${global_basedir}/eblogs/report.log.${ds_ds_run} 1>/dev/null 2>&1\r\n\r\n\r\nalarm_0201\r\n\r\n\r\nglobal_report_new=`cat ${global_basedir}/eblogs/report.log|sed ':a;N;$!ba;s/\\n/|/g'`\r\nif [ X\"${mgr_out_report}\" != X ]\r\nthen\r\n    global_report=\"${mgr_out_report}|${global_report_new}\"\r\nelse\r\n    global_report=\"${global_report_new}\"\r\nfi\r\nglobal_report=`cat ${global_basedir}/eblogs/report.log|sed ':a;N;$!ba;s/\\n/|/g'`\r\n\r\necho '${setValue(mgr_out_report='${global_report}')}'\r\necho '${setValue(continue_flag=continue)}'\r\n\r\nbad_num=`cat ${global_basedir}/eblogs/report.log|grep fail|wc -l|awk '{print $1}'`\r\nif [ ${bad_num} -gt 0 ]\r\nthen\r\n    exit 1\r\nfi","resourceList":[{"id":31,"resourceName":"/public.sh","res":"public.sh"},{"id":32,"resourceName":"/alarm/testInMGR.sh","res":"testInMGR.sh"}]}
[INFO] 2022-09-06 04:50:45.841 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[139] - raw script : set -x
if [ X"null" != Xstart ] 
then
    if [ X"null" != Xcontinue ]
    then
        global_report="${mgr_out_report}"
        echo '${setValue(mgr_out_report='${global_report}')}'
        exit 0
    else
        global_report=""
    fi
fi
basedir=/home/dolphinscheduler
source ./public.sh
source ./alarm/testInMGR.sh
mv /home/dolphinscheduler/eblogs/report.log  /home/dolphinscheduler/eblogs/report.log.20220906045045 1>/dev/null 2>&1

alarm_0201

global_report_new=`cat /home/dolphinscheduler/eblogs/report.log|sed ':a;N;$!ba;s/\n/|/g'`
if [ X"${mgr_out_report}" != X ]
then
    global_report="${mgr_out_report}|${global_report_new}"
else
    global_report="${global_report_new}"
fi
global_report=`cat /home/dolphinscheduler/eblogs/report.log|sed ':a;N;$!ba;s/\n/|/g'`

echo '${setValue(mgr_out_report='${global_report}')}'
echo '${setValue(continue_flag=continue)}'

bad_num=`cat /home/dolphinscheduler/eblogs/report.log|grep fail|wc -l|awk '{print $1}'`
if [ ${bad_num} -gt 0 ]
then
    exit 1
fi
[INFO] 2022-09-06 04:50:45.841 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[140] - task execute path : /tmp/dolphinscheduler/exec/process/6585171189344/6773442559590_13/341/617
[INFO] 2022-09-06 04:50:45.841 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[85] - tenantCode user:root, task dir:341_617
[INFO] 2022-09-06 04:50:45.842 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[90] - create command file:/tmp/dolphinscheduler/exec/process/6585171189344/6773442559590_13/341/617/341_617.command
[INFO] 2022-09-06 04:50:45.842 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[116] - command : #!/bin/sh
BASEDIR=$(cd `dirname $0`; pwd)
cd $BASEDIR
source /home/dolphinscheduler/worker-server/conf/dolphinscheduler_env.sh
/tmp/dolphinscheduler/exec/process/6585171189344/6773442559590_13/341/617/341_617_node.sh
[INFO] 2022-09-06 04:50:45.844 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[290] - task run command: sudo -u root sh /tmp/dolphinscheduler/exec/process/6585171189344/6773442559590_13/341/617/341_617.command
[INFO] 2022-09-06 04:50:45.844 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[181] - process start, process id is: 130117
[INFO] 2022-09-06 04:50:45.866 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[205] - process has exited, execute path:/tmp/dolphinscheduler/exec/process/6585171189344/6773442559590_13/341/617, processId:130117 ,exitStatusCode:0 ,processWaitForStatus:true ,processExitValue:0
[ERROR] 2022-09-06 04:50:45.866 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[104] - shell task error
java.lang.ArrayIndexOutOfBoundsException: 1
	at org.apache.dolphinscheduler.plugin.task.api.parameters.AbstractParameters.getMapByString(AbstractParameters.java:191)
	at org.apache.dolphinscheduler.plugin.task.api.parameters.AbstractParameters.dealOutParam(AbstractParameters.java:144)
	at org.apache.dolphinscheduler.plugin.task.shell.ShellTask.handle(ShellTask.java:102)
	at org.apache.dolphinscheduler.server.worker.runner.TaskExecuteThread.run(TaskExecuteThread.java:208)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[INFO] 2022-09-06 04:50:45.867 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[63] -  -> ++ '[' Xnull '!=' Xstart ']'
	++ '[' Xnull '!=' Xcontinue ']'
	++ global_report=
	++ echo '${setValue(mgr_out_report=)}'
	++ exit 0
[INFO] 2022-09-06 04:50:45.867 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[57] - FINALIZE_SESSION
[INFO] 2022-09-06 04:50:45.867 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[230] - cancel process: 130117
[INFO] 2022-09-06 04:50:46.844 +0000 [taskAppId=TASK-20220906-6773442559590_13-341-617] TaskLogLogger-class org.apache.dolphinscheduler.plugin.task.shell.ShellTask:[57] - FINALIZE_SESSION

**Comment by DarkAssassinator** [comment]
please check shell's result format:
![image](https://user-images.githubusercontent.com/20518339/188668031-ac3d5353-fc10-419c-bcd7-6875db0bdb1d.png)
